### ISSUE TITLE
System.Diagnostics.Eventing.Reader.EventLogRecord throws an InvalidOperationException on Windows 2022 Server with .NET 4.8 runtime

### ISSUE THREAD

**Issue by reveillesoftware** [dotnet/runtime#79515]
### Discussed in https://github.com/dotnet/runtime/discussions/79510

<div type='discussions-op-text'>

<sup>Originally posted by **reveillesoftware** December 11, 2022</sup>
Receive this error on Windows 2022 Server with all current Windows Server updates:  ERROR : We do not have 18 variants given for the UnsafeNativeMethods.EvtRenderFlags.EvtRenderEventValues flag. (System Properties)<br /> at System.Diagnostics.Eventing.Reader.NativeWrapper.EvtRenderBufferWithContextSystem(EventLogHandle contextHandle, EventLogHandle eventHandle, EvtRenderFlags flag, SystemProperties systemProperties, Int32 SYSTEM_PROPERTY_COUNT) at System.Diagnostics.Eventing.Reader.EventLogRecord.PrepareSystemData() at System.Diagnostics.Eventing.Reader.EventLogRecord.get_TimeCreated() at ReveilleAPI.EventLogging.Client.CheckNewEventLog(String inComputerName, String inLog, String inSourceName, String inEventIDs, EventLogEntryTypeEx inEventType, Int32 inSecondsOld, String separator, Boolean& anyMessagesFound)

This was fixed in last year for WIndows 11, when will this be ported for Windows server 2022?  

See - https://github.com/dotnet/runtime/issues/60740

This prevents monitoring of the Windows Event log in Windows Server 2022 by 3rd party tools that use Windows API.</div>

**Comment by ViktorHofer** [comment]
from @Robo210:

> I can't personally provide any official support nor details about internal decisions. I can say, however, that if you have an avenue to make formal support requests to Microsoft, please do so and let them know that you are still waiting for this fix.

@reveillesoftware can you please file an official support ticket? We don't have any control over when a corresponding patch will be released for Windows Server 2022.

**Comment by reveillesoftware** [comment]
Yes, we did file a MSFT support ticket via MSFT Azure team and they then indicated to post here.  What is the proper channel to open an official support ticket?

**Comment by ViktorHofer** [comment]
> What is the proper channel to open an official support ticket?

@ericstj @danmoseley do you maybe know?

**Comment by danmoseley** [comment]
Through standard product support channels.

https://support.microsoft.com/en-us

@robo210 do you have any update?

**Comment by reveillesoftware** [comment]
Ok, we just looked at that support channel and it is asking for $499 US to create a .NET framework support case.  Given this is a known defect with a known solution that needs to be ported to Windows 2022, interesting it requires payment to simply open a separate product support case given we already pay for Azure developer support, Visual Studio plans, Microsoft Partner portal subscriptions, etc.

Thank You for the assistance.

**Comment by danmoseley** [comment]
@reveillesoftware yeah, fair point. Reaching out internally to Windows folks.

**Comment by reveillesoftware** [comment]
@danmoseley -thank you, appreciate if you receive feedback from your reach out.

**Comment by danmoseley** [comment]
What I learned is that it's being considered for servicing at the moment but there's no decision or date to share. And there's no need to do anything special to +1 it. Posting or thumbs up on this issue is enough. I'll update here promptly if I learn more that I can share.
